diff --git a/dist/esbuild-plugin-astro-frontmatter.js b/dist/esbuild-plugin-astro-frontmatter.js
index 7a6eb81f94c629aa93a4105ccedb53135ac93f12..0704fd41468f1f03d8413015163fb1bf446a204a 100644
--- a/dist/esbuild-plugin-astro-frontmatter.js
+++ b/dist/esbuild-plugin-astro-frontmatter.js
@@ -4,21 +4,28 @@ function astroFrontmatterScanPlugin() {
   return {
     name: "astro-frontmatter-scan",
     setup(build) {
-      build.onLoad({ filter: /\.astro$/ }, async (args) => {
+      const extractFrontmatter = async (path) => {
         try {
-          const code = await readFile(args.path, "utf-8");
+          const code = await readFile(path, "utf-8");
           const frontmatterMatch = FRONTMATTER_RE.exec(code);
           if (frontmatterMatch) {
-            const contents = frontmatterMatch[1].replace(/\breturn\s*;/g, "throw 0;").replace(/\breturn\b/g, "throw ");
-            return {
-              contents,
-              loader: "ts"
-            };
+            return frontmatterMatch[1].replace(/\breturn\s*;/g, "throw 0;").replace(/\breturn\b/g, "throw ");
           }
         } catch {
         }
+        return "";
+      };
+      build.onLoad({ filter: /\.astro$/, namespace: "file" }, async (args) => {
         return {
-          contents: "",
+          contents: await extractFrontmatter(args.path),
+          loader: "ts"
+        };
+      });
+      build.onLoad({ filter: /\.astro$/, namespace: "html" }, async (args) => {
+        const contents = await extractFrontmatter(args.path);
+        return {
+          contents: `${contents}
+export default {};`,
           loader: "ts"
         };
       });
